libvast/src/error.cpp

```diff
@@ -11,45 +11,94 @@
  * contained in the LICENSE file.                                             *
  ******************************************************************************/
 
+#include <sstream>
+
+#include <caf/pec.hpp>
+#include <caf/sec.hpp>
+
 #include "vast/detail/assert.hpp"
 #include "vast/error.hpp"
 
 namespace vast {
 namespace {
 
 const char* descriptions[] = {
-  "no_error",
+  "no error",
   "unspecified",
-  "no_such_file",
-  "filesystem_error",
-  "type_clash",
-  "unsupported_operator",
-  "parse_error",
-  "print_error",
-  "convert_error",
-  "invalid_query",
-  "format_error",
-  "end_of_input",
-  "version_error",
-  "syntax_error",
-  "invalid_table_slice_type",
-  "invalid_synopsis_type",
-  "remote_node_down",
-  "invalid_result",
-  "invalid_configuration",
-  "unrecognized_option",
-  "invalid_subcommand",
-  "missing_subcommand",
-  "no_importer",
+  "no such file",
+  "filesystem error",
+  "type clash",
+  "unsupported operator",
+  "parse error",
+  "print error",
+  "convert error",
+  "invalid query",
+  "format error",
+  "end of input",
+  "version error",
+  "syntax error",
+  "invalid table slice type",
+  "invalid synopsis type",
+  "remote node down",
+  "invalid result",
+  "invalid configuration",
+  "unrecognized option",
+  "invalid subcommand",
+  "missing subcommand",
+  "no importer",
   "unimplemented",
 };
 
-} // namespace <anonymous>
+void render_vast_default_ctx(std::ostringstream& oss, const caf::message& ctx) {
+  size_t size = ctx.size();
+  if (size > 0) {
+    oss << ":";
+    for (size_t i = 0; i < size; ++i)
+      oss << " \"" << ctx.get_as<std::string>(i) << '"';
```

Comment:
There's actually no need to assume strings here. [`ctx.stringify(i)`](https://github.com/actor-framework/actor-framework/blob/537de9e681325eb146e3b13e5caed92615e48bb3/libcaf_core/caf/type_erased_tuple.hpp#L77) is safer and more concise. Also, this allows you to combine the two functions `render_vast_default_ctx` and `render_default_ctx ` into a single one.